tests/test_views.py

```diff
@@ -0,0 +1,137 @@
+"""Tests for views are at this file."""
+from unittest.mock import patch
+from datetime import datetime
+
+from django.test import TestCase, Client
+from django.contrib.auth.models import User
+from django.urls import reverse
+from django.utils import timezone
+
+from blog.models import Post
+
+
+class ViewsTest(TestCase):
+    """Testing class for views."""
+
+    USERNAME = 'admin'
+    PASSWORD = '12345678'
+
+    def setUp(self):
+        """Prepare data for testing."""
+        self.client = Client()
+        self.user = User.objects.create(
+            username='testuser',
+            password='blablabla',
+            is_superuser=True, is_staff=True,
+            email='testuser@gmail.com',
+            is_active=True,
+        )
+
+    def test_index_view_rendering(self):
+        """Testing main page for render needed posts."""
+        time_zone = timezone.get_current_timezone()
+        future_post = Post.objects.create(
+            author=self.user,
+            title='future_test',
+            text='superText',
+            created_date=datetime(
+                day=1, month=4,
+                year=3020,
+                tzinfo=time_zone,
+            ),
+            published_date=datetime(
+                day=1,
+                month=4,
+                year=3020,
+                tzinfo=time_zone,
+            ),
+        )
+        post = Post.objects.create(
+            author=self.user, title='Test',
+            text='superText',
+            created_date=datetime(
+                day=1, month=3,
+                year=2020, tzinfo=time_zone,
+            ),
+            published_date=datetime(
+                day=1, month=3,
+                year=2020,
+                tzinfo=time_zone,
+            ),
+        )
+        past_post = Post.objects.create(
+            author=self.user, title='past_est',
+            text='superText',
+            created_date=datetime(
+                day=1, month=4,
+                year=2019,
+                tzinfo=time_zone,
+            ),
+            published_date=datetime(
+                day=1, month=4,
+                year=2019,
+                tzinfo=time_zone,
+            ),
+        )
+        with patch(
+                'django.utils.timezone.now', lambda: datetime(
+                    day=1,
+                    month=1,
+                    year=2020,
+                    tzinfo=time_zone,
+                ),
+        ):
+            response = self.client.get(reverse('post_list'))
+            for _ in response:
+                with self.subTest():
+                    self.assertListEqual(
+                        list(response.context['posts']), [
+                            past_post,
+                        ],
+                    )
+                    self.assertNotContains(response, post)
+                    self.assertNotContains(response, future_post)
```

Comment:
The last two checks are redundant: if the list is equal to having just the past post, it's implied that the present and the future ones are not there by the first check.